### PR TITLE
Copter: do not go to min throttle mix with failed motor

### PR DESCRIPTION
**Pull request by IamPete1**
Going to min throttle mix with a failed motor can lead to very poor attitude control.

### CODE REVIEW

**Comment by IamPete1** [comment]
I'm not sure if we should only allow this if not landing like the large angle request check.

**Comment by lthall** [comment]
Can I see the log?

**Comment by IamPete1** [comment]
@lthall Rebased on forced flying and only applying if not landing.

**Comment by lthall** [comment]
The problem is this then would not fix the problem you saw because it happened when landing started. That is why it went to mix min in the first place.

Maybe we can never go to mix min but not make it a requirement in the land detector.

**Comment by IamPete1** [comment]
@lthall My thinking was that this would at least be better than nothing, and is not a major change that would require such extensive testing. Your right that in the case of the that log it would not have helped.

We could do as you say, but thrust loss is quite common false positive on some vehicles, I guess the flag should clear quickly tho. At least we have `_thrust_boost` logged in `MOTB` now so we should get more insight into what is going on in the future.   

Basically I'm a bit scared of unintended consequences in the landing detector.

**Comment by lthall** [comment]
Yes, I agree. This little file is one of the trickiest parts of the code and so easy to cause a crash (or a series of crashes, been there done that).

>> My thinking was that this would at least be better than nothing.

I am not so sure. This would remove the ability of any aircraft suffering a motor failure to detect a landing in an autonomous mode or failsafe. This is in an attempt to reduce the chance an aircraft that is only marginally maintaining control from crashing, and may crash anyway.

>> At least we have _thrust_boost logged in MOTB now so we should get more insight into what is going on in the future.
>> Basically I'm a bit scared of unintended consequences in the landing detector.

Yes I agree on both counts.

My thoughts here are if we can't fix the actual problem we should not change the code. This is one of the scariest parts of the code for me, so many incidents have been caused by slightly off decisions here.

**Comment by IamPete1** [comment]
> My thoughts here are if we can't fix the actual problem we should not change the code. This is one of the scariest parts of the code for me, so many incidents have been caused by slightly off decisions here.

Although we don't fix the problem completely, we do delay it. Much better to fall out of the sky somewhere your expecting to land than in some other part of the mission. At least (one hopes) there will be a clear area and some expectation of the dangerous thing descending imminently. 

This might also give enough time to for the pilot react to the thrust loss warning and recover in a manual mode.  

Of course we cannot be sure that its safer to crash over landing point than at some other point in every case. I guess there is some argument that currently we do not make a choice, this change means that we have actively made that choice whether or not it is the correct one in most cases.

**Comment by IamPete1** [comment]
I guess we don't actually have to decide, we would enable by a param. Its the sort of thing users only find after the crash tho. Might help on the next crash I guess.....

**Comment by lthall** [comment]
Adding code that may or may not fix a problem without understanding the problem is a bad path to go down. 

1. From a mathematical perspective the reduction of throttle should not cause an aircraft to loose control.
2. I am aware that the aircraft that experienced these issues has a number of control problems and unusual characteristics that mean the tune is significantly compromised.
3. This makes if very hard to use this incident to justify removing the ability for aircraft to detect landing, especially as this change would not make any difference to the incident that prompted it.
4. In the end we are left with a significant change and loss of functionality with no justification.
5. Then adding a parameter to get past this point when we can't tell anybody how to decide how to set that parameter or even say under what circumstances it would help.......

I have a development plan I have talked about a few times on the dev call that goes something like:

- remove loiter and wp_nav from take-off and landing (the goal is to simplify the code making it easer to deal with initialisation and relaxation of control systems)
- Add ground -> air transition handling
- Make better use of "land maybe" to make the air -> ground transition safer

This last part may provide us additional options to delay some state changes to a later time and reduce risk.

Apart from the incident that prompted this change (an aircraft with some problems from a tuning perspective), X8 aircraft deal with lost motors with almost zero change in control. On the other hand, Hex aircraft are a nightmare to deal with. I am also hoping to get some time to do some hard core motor out testing on the Callisto 50 to develop a reference data set on a known good aircraft. This may help us define the lines with a little more confidence.

I think we are better off moving slowly and deliberately in this part of the code with very clear and through justification for changes.
